odilo.py

```diff
@@ -0,0 +1,581 @@
+            extra_headers = {}
+        headers = dict(Authorization="Bearer %s" % self.token)
+        headers.update(extra_headers)
+        status_code, headers, content = self._do_get(url, headers)
+        if status_code == 401:
+            if exception_on_401:
+                # This is our second try. Give up.
+                raise BadResponseException.from_response(
+                    url,
+                    "Something's wrong with the Odilo OAuth Bearer Token!",
+                    (status_code, headers, content)
+                )
+            else:
+                # Refresh the token and try again.
+                self.check_creds(True)
+                return self.get(url, extra_headers, True)
+        else:
+            return status_code, headers, content
+
+    def token_post(self, url, payload, headers={}, **kwargs):
+        """Make an HTTP POST request for purposes of getting an OAuth token."""
+        s = "%s:%s" % (self.client_key, self.client_secret)
+        auth = base64.standard_b64encode(s).strip()
+        headers = dict(headers)
+        headers['Authorization'] = "Basic %s" % auth
+        headers['Content-Type'] = "application/x-www-form-urlencoded"
+        return self._do_post(url, payload, headers, **kwargs)
+
+    @staticmethod
+    def _update_credential(credential, odilo_data):
+        """Copy Odilo OAuth data into a Credential object."""
+        credential.credential = odilo_data['token']
+        expires_in = (odilo_data['expiresIn'] * 0.9)
+        credential.expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=expires_in)
+
+    def get_metadata(self, record_id):
+        identifier = record_id
+        if isinstance(record_id, Identifier):
+            identifier = record_id.identifier
+
+        url = self.RECORD_METADATA_ENDPOINT.replace('{recordId}', identifier)
+
+        response = self.get(url)
+
+        if response and response.status == 200 and response.content:
+            return response.content
+        else:
+            self.log.warn('Cannot retrieve metadata for record: ' + record_id + ' response http ' + response.status)
+            if response.content:
+                self.log.warn(response.content)
+            return None
+
+    def get_availability(self, record_id):
+        url = self.RECORD_AVAILABILITY_ENDPOINT.replace('{recordId}', record_id)
+        status_code, headers, content = self.get(url)
+        content = json.loads(content)
+
+        if status_code == 200 and len(content) > 0:
+            return content
+        else:
+            self.log.warn('Cannot retrieve availability for record: ' + record_id + ' response http ' + status_code)
+            if content:
+                self.log.warn(content)
+            return None
+
+    @staticmethod
+    def _do_get(url, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
```

Comment:
kwargs is never going to be null, so you don't need the first clause. However if a specific value is passed in for `timeout` I would use that value instead of forcing it to 60. (the value passed in may be bigger than 60).

Reply:
ok